plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/cleaner/DockerAbandonedResourcesCleaner.java

```diff
@@ -15,52 +15,88 @@
 
 import org.eclipse.che.api.core.NotFoundException;
 import org.eclipse.che.api.environment.server.CheEnvironmentEngine;
+import org.eclipse.che.api.workspace.server.WorkspaceRuntimes;
 import org.eclipse.che.commons.schedule.ScheduleRate;
 import org.eclipse.che.plugin.docker.client.DockerConnector;
 import org.eclipse.che.plugin.docker.client.json.ContainerListEntry;
+import org.eclipse.che.plugin.docker.client.json.Filters;
+import org.eclipse.che.plugin.docker.client.json.network.Network;
+import org.eclipse.che.plugin.docker.client.params.network.GetNetworksParams;
 import org.eclipse.che.plugin.docker.machine.DockerContainerNameGenerator;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.inject.Named;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.Set;
 import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
 import static org.eclipse.che.plugin.docker.client.params.RemoveContainerParams.create;
 import static org.eclipse.che.plugin.docker.machine.DockerContainerNameGenerator.ContainerNameInfo;
 
 /**
- * Job for periodically clean up inactive docker containers and log list active containers.
+ * Job for periodically clean up abandoned docker containers and networks created by CHE.
+ * Also, logs active containers list.
  *
  * @author Alexander Andrienko
+ * @author Mykola Morhun
  */
 @Singleton
-public class DockerContainerCleaner implements Runnable {
+public class DockerAbandonedResourcesCleaner implements Runnable {
 
-    private static final Logger LOG = LoggerFactory.getLogger(DockerContainerCleaner.class);
+    private static final Logger LOG = LoggerFactory.getLogger(DockerAbandonedResourcesCleaner.class);
+
+    private static final String            CHE_NETWORK_NAME_PREFIX  = "workspace";
+    private static final Filters           NETWORK_FILTERS          = new Filters().withFilter("type", "custom");
+                                                                                  // TODO name filter doesn't work with pure docker swarm
```

Comment:
It is not used now and I don't think it is useful enough to leave it as a comment